front/src/Phaser/Game/GameMapPropertyChange.ts

```diff
@@ -0,0 +1,55 @@
+import type { GameScene } from "./GameScene";
+import type { GameMap } from "./GameMap";
+import { scriptUtils } from "../../Api/ScriptUtils";
+import { coWebsiteManager } from "../../WebRtc/CoWebsiteManager";
+import { layoutManagerActionStore } from "../../Stores/LayoutManagerStore";
+import {
+    ON_ACTION_TRIGGER_BUTTON,
+    TRIGGER_WEBSITE_PROPERTIES,
+    WEBSITE_MESSAGE_PROPERTIES,
+} from "../../WebRtc/LayoutManager";
+
+export class GameMapPropertyChange {
+    constructor(private scene: GameScene, private gameMap: GameMap) {}
+
+    register() {
```

Comment:
you might as well just put all this code directly into the constructor

Reply:
Also you could put all the other this.gameMap.onPropertyChange from GameScene here, and maybe put every listened properties into constants, that way we would have an easy to list map properties

Reply:
well if i did that we'd have to make some properties public that were private before 
personally id also prefer if the implementation of the listeners was somewhere external - to basically have a single easily graspable file that just has one line for each listener

Reply:
i dont think typescript likes just calling new () without then using the returned value

Reply:
how about we just convert the entire calss to a function ?

Reply:
but the gameScene class is already waaaay too big and refactorings will usally only be done when christmas is on the same day as easter (never :D )

Reply:
But I like to do big refactorings :( . But you're not wrong and the GameScene is very tricky to edit, I am just not a that to create random class that don't really have a well contained purpose. I'll guess I'll allow it this time because this PR is usefull